### DISCUSSION TITLE
Output a function of the solution vtk

### DISCUSSION THREAD

**Discussion by cjham** [firedrakeproject/firedrake#3847] [Firedrake support] [answered]
Hello, Sorry this is probably a very basic question but I haven't been able to get it to work. I'd like to output the solution, u, to vtk which I can do but I'd also like to output the derivative in the x direction of the solution. I am not sure how to make this happen. I have done various things with Dx(u,0) but none have worked!

**Comment by connorjward** [comment, marked as answer]
I think your issue is likely that `Dx(u, 0)` is a *symbolic expression* whereas for visualisation you want a `Function`. I think you can get what you want by doing
```py
deriv = Function(V).interpolate(Dx(u, 0))

# now plot deriv
```

**Reply by cjham** [reply]
This worked, thank you.
